### DISCUSSION TITLE
TTS removed models

### DISCUSSION THREAD

**Discussion by amirrezam** [NVIDIA-NeMo/NeMo#4178] [General]
Hi,
Why Some TTS model (MelGAN, TalkNet) are removed from main branch, while they are exist in 1.8.2 ? 
Do you want to stop maintaining them ?

https://github.com/NVIDIA/NeMo/commit/c37786a9acb443b1d55d56636f00d3826ffc76d5

**Comment by redoctopus** [comment]
We recently unified the TTS models' data loading under `TTSDataset`, which breaks support for older checkpoints that depend on model-specific Dataset classes. As a result, we've decided to deprecate some older and lower-quality models to focus on and update a smaller set of flagship models.

**Reply by ahkarami** [reply]
@redoctopus 
Thanks for your answer. If one trains previous models (e.g., FastSpeech2) via previous versions of NeMo, then for the test time (inference time), he can use this model via the newer versions of NeMo?
I mean that, unifying the data loader class doesn't affect the inference time? Am I correct?
Best

**Reply by ahkarami** [reply]
Just as another important note. I think it is better to maintain at least `FastSpeech2` & `MelGAN` models. Because these are classic and important models.
Best

**Reply by redoctopus** [reply]
The older dataset classes as well as the model classes have been removed, so 1.8.2 is the last version of NeMo that supports the old checkpoints.

Ideally we'd like to support all our previous models, but at this time we don't have enough engineer hours to convert, re-train, and maintain all of them, unfortunately :( Thanks for the feedback, though, it's useful for prioritization in the future!

**Reply by ahkarami** [reply]
Thanks for your answer. I hope to see the growing success of your team in the near future.
Best
